modules/devices.nix

```diff
@@ -109,14 +109,17 @@ in lib.mkMerge [{
   # On Xavier AGX, setting UEFI variables requires having the ESP partition on the eMMC:
   # https://forums.developer.nvidia.com/t/setting-uefi-variables-using-the-defaultvariabledxe-only-works-if-esp-is-on-emmc-but-not-on-an-nvme-drive/250254
   # We don't mount this at /boot, because we still want to allow the user to have their ESP part on NVMe, or whatever else.
-  fileSystems."/opt/nvidia/esp" = lib.mkDefault {
-    device = "/dev/disk/by-partlabel/esp";
-    fsType = "vfat";
-    options = [ "nofail" ];
-    # Since we have NO_ESP_IMG=1 while formatting, the script doesn't
-    # actually create an FS here, so we'll do it automatically
-    autoFormat = true;
-    formatOptions = "-F 32 -n ESP";
-  };
+  fileSystems."/opt/nvidia/esp" = lib.mkDefault ({ config, ... }: {
+    config = {
+      device = "/dev/disk/by-partlabel/esp";
+      fsType = "vfat";
+      options = [ "nofail" ];
+      # Since we have NO_ESP_IMG=1 while formatting, the script doesn't
+      # actually create an FS here, so we'll do it automatically
+      autoFormat = true;
+      # needed on nixpkgs < 23.11
+      formatOptions = if (lib.any (opt: opt == "x-systemd.makefs") config.options) then null else "-F 32 -n ESP";
```

Comment:
Is this just to condition on NixOS version? The better way would be to condition on `config.system.nixos.release`